LineChart.js

```diff
@@ -0,0 +1,1161 @@
+
+            var confirm = d3.select('.confirm')
+                .style('visibility', 'visible');
+
+            var _filter = [];
+            data.forEach(function (d) {
+                var obj = new Object();
+                obj[_dimension[0]] = d[_dimension[0]];
+                for (var index = 0; index < _measure.length; index++) {
+                    obj[_measure[index]] = d[_measure[index]];
+                }
+
+                _filter.push(obj)
+            });
+            if (_filter.length > 0) {
+                filterData = _filter;
+            }
+        }
+    }
+    var applyFilter = function (chart) {
+        return function () {
+            if (filterData.length > 0) {
+                chart.update(filterData);
+            }
+        }
+    }
+    var clearFilter = function () {
+        return function () {
+            chart.update(_originalData);
+        }
+    }
+    var _handleMouseOverFn = function (tooltip, container) {
+        var me = this;
+
+        return function (d, i) {
+            d3.select(this).style('cursor', 'pointer')
+                .style('cursor', 'pointer')
+                .style('fill', COMMON.HIGHLIGHTER);
+            var border = UTIL.getDisplayColor(_measure.indexOf(d.tag), _displayColor)
+            if (tooltip) {
+                UTIL.showTooltip(tooltip);
+                UTIL.updateTooltip.call(tooltip, _buildTooltipData(d, me), container, border);
+            }
+        }
+    }
+    var _handleMouseMoveFn = function (tooltip, container) {
+        var me = this;
+
+        return function (d, i) {
+            if (tooltip) {
+                var border = UTIL.getDisplayColor(_measure.indexOf(d.tag), _displayColor)
+                UTIL.updateTooltip.call(tooltip, _buildTooltipData(d, me, border), container, border);
+            }
+        }
+    }
+    var _handleMouseOutFn = function (tooltip, container) {
+        var me = this;
+
+        return function (d, i) {
+            d3.select(this).style('cursor', 'default');
+
+            var arcGroup = container.selectAll('g.arc')
+                .filter(function (d1) {
+                    return d1.data[_dimension[0]] === d.data[_dimension[0]];
+                });
+
+            arcGroup.select('path')
+                .style('fill', function (d1, i) {
+                    return COMMON.COLORSCALE(d1.data[_dimension[0]]);
+                });
+
+            var arcMaskGroup = container.selectAll('g.arc-mask')
+                .filter(function (d1) {
+                    return d1.data[_dimension[0]] === d.data[_dimension[0]];
+                });
+
+            arcMaskGroup.select('path')
+                .style('visibility', 'hidden');
+
+            if (tooltip) {
+                UTIL.hideTooltip(tooltip);
+            }
+        }
+    }
+    function chart(selection) {
+        _local_svg = selection;
+
+        selection.each(function (data) {
+            chart._Local_data = _originalData = data;
```

Comment:
Let's not place the data or other information in a publicly accessible property directly.